src/FilterChain.php

```diff
@@ -149,6 +149,15 @@ public function plugin($name, array $options = [])
      */
     public function attach($callback, $priority = self::DEFAULT_PRIORITY)
     {
+        if (is_object($callback) && ! $callback instanceof \Closure) {
+            if (! $callback instanceof FilterInterface) {
```

Comment:
Why a second condition?

Reply:
You are right, there is no need in this case. I collapsed them in one.